airbyte-queue/src/main/java/io/airbyte/queue/BigQueue.java

```diff
@@ -111,4 +146,48 @@ public void close() throws Exception {
     queue.gc();
   }
 
+  /**
+   * Print size instead of queue contents to avoid any sort of logging complication. Note this does
+   * not hold any read locks for simplicity, and queue size cannot be used as a source of truth.
+   */
+  @Override
+  public String toString() {
+    return "BigQueue{" +
+        "queue=" + queue.hashCode() +
+        ", size=" + queue.size() +
+        ", closed=" + closed +
+        '}';
+  }
+
+  private interface NonBlockingOp {
```

Comment:
`NonBlockingOp` / `BlockingOp` aren't very descriptive imo and are identical in terms of a specific `Lock`. Really they're just safely locking around a block.

We could instead have:
```
  private class LockedSupplier<T> {

    private final Lock lock;
    private final Supplier<T> delegate;

    public LockedSupplier(Lock lock, Supplier<T> delegate) {
         this.lock = lock;
         this.delegate = delegate;
     }

    static <T> T get(Lock lock, Supplier<T> supplier) {
      lock.lock();
      try {
        return supplier.get();
      } catch (Exception e) {
        throw new RuntimeException(e);
      } finally {
        lock.unlock();
      }
    }

  }
```

And then we would call it with:
```
    return new LockedSupplier(lock.writeLock(), () -> {
      try {
        List<byte[]> elements = new ArrayList<>();
        while (!queue.isEmpty()) {
          elements.add(queue.dequeue());
        }
        for (byte[] e : elements) {
          queue.enqueue(e);
        }
        return elements.iterator();
      } catch (IOException e) {
        throw new RuntimeException(e);
      }
    }).get();
```

or similar.